server/public/model/user.go

```diff
@@ -703,14 +707,14 @@ func (u *User) SanitizeInput(isAdmin bool) {
 
 func (u *User) ClearNonProfileFields(asAdmin bool) {
 	u.Password = ""
-	u.AuthData = NewPointer("")
 	u.MfaSecret = ""
 	u.MfaUsedTimestamps = nil
 	u.EmailVerified = false
 	u.AllowMarketing = false
 	u.LastPasswordUpdate = 0
 
 	if !asAdmin {
+		u.AuthData = NewPointer("")
```

Comment:
Is this needed? A user may be an system admin but may not have mmctl access.

Reply:
Yes. With this change, a system admin can retrieve AuthData of a user, while this is only meant to be for local mode via mmctl.

Let me know if that's more clear or if I've misunderstood something

Reply:
The `mmctl user search` command is not only available in local mode, but also via HTTP. Is that concerning?

Reply:
Nope, I thought it was only via local mode.